scripts/header_parser.py

```diff
@@ -433,13 +449,119 @@ def generate_dot_Call_wrapper(func_name: str, params_list: list[Parameter]) -> s
     )
 
 
-def create_wrappers_for_header_file(path: str,
+def generate_C_wrapper(func_name: str, params_list: list[Parameter], return_type: str) -> str:
+    """
+    Uses a function name, its return type and its parameters (a list of
+    :class:`Parameter`) to create a C wrapper to a C function using R's SEXP
+    type.
+
+    Example: the function defined as follows::
+
+        /**
+         * calculates time step for balancing pressure in pipes
+         * @\0param n  number of items
+         * @\0param xs numbers to take mean of (array of length n)
+         * @\0return harmonic mean of xs
+         */
+        double harmonic_mean(const int n, const double *xs);
+
+    Will produce a wrapper like this::
+
+        SEXP harmonic_mean_wrapper(
+            SEXP p_n,
+            SEXP p_xs,
+        ){
+            SEXP RETVAL = PROTECT(AllocVector(REALSXP, 1));
+            int n = *(INTEGER(p_n));
+            double* xs = (double*)malloc(sizeof(double) * n);
+            for(int i = 0; i < n; i++) xs[i] = (REAL(p_xs))[i];
+            *(REAL(RETVAL)) = harmonic_mean(n, xs);
+            free(xs);
+            UNPROTECT(1);
+            return RETVAL;
+        }
+    
+    :param func_name: name of the base function
+    :param params_list: list of :class:`Parameter`, parameters of ``func_name``
+    :param return_type: return type of the base function, e.g. ``unsigned int``
+    :return: a C wrapper compatible with the .Call interface
+    """
+    return_ctype = str_to_ctype(return_type)
+    signature = [f'SEXP {func_name}_wrapper(']
+    middle = [
+        '){',
+        '\tSEXP RETVAL = PROTECT(AllocVector({}, 1));'
+        .format(type_registration[return_ctype])
+    ]
+    late = []
+    call = ['\t*({}(RETVAL)) = {}({});'
+            .format(SEXP_conversion[return_ctype],
+                    func_name, ', '.join(f'{p.name}' for p in params_list))]
+    end = ['\tUNPROTECT(1);', '\treturn RETVAL;', '}']
+    for param in params_list:
+        signature.append(f'\tSEXP p_{param.name},')
+        if param.size is None:
+            if (not param.raw_type.endswith('*')
```

Comment:
I'd suggest to remove brackets inside if and merge the two lines below into single line